### PR TITLE
Allow audio file transcription using deepgram, whisper api, whisper.cpp

### PR DESCRIPTION
**Pull request by vivekuppal**
Based on user request

### CODE REVIEW

**Comment by techmemes** [comment]
This is helpful. Thank You!

File: app/transcribe/audio_player.py [hunk 1]
```diff
@@ -72,7 +72,11 @@ def play_audio_loop(self, config: dict):
     def _get_language_code(self, lang: str) -> str:
         """Get the language code from the configuration.
         """
-        return next(key for key, value in LANGUAGES_DICT.items() if value == lang)
+        try:
```
Comment (mang0sw33t) [inline app/transcribe/audio_player.py:75]: Good catch

File: app/transcribe/app_utils.py [hunk 2]
```diff
@@ -192,6 +196,17 @@ def create_transcriber(
     global_vars.set_transcriber(t)
 
 
+def get_language_code(lang: str) -> str:
```
Comment (mang0sw33t) [inline app/transcribe/app_utils.py:199]: I think this method is repeated. Not a big deal, can be taken care of in a subsequent PR.

File: app/transcribe/audio_transcriber.py [hunk 3]
```diff
@@ -228,9 +228,10 @@ def convert_wav_to_16khz_format(self, file_path: str) -> str:
             file_descritor, mod_file_path = tempfile.mkstemp(suffix=".wav")
             os.close(file_descritor)
             # print(f'Convert file {file_path} to 16khz file {mod_file_path}')
+            log_file = f"{utilities.get_data_path(app_name='Transcribe')}/logs/ffmpeg.txt"
```
Comment (mang0sw33t) [inline app/transcribe/audio_transcriber.py:231]: Good catch that these log files were not being produced in the correct location.

**Review by mang0sw33t — APPROVED** [review]
Thanks for the offline discussion!

Good enhancements with small changes. Nice work, goes to show that the code is well structured when we can add significant enhancements with small changes in a few places.

Nice work!

**Review by j0n0w1ns — APPROVED** [review]

**Pull request merged**